File: examples/invoke_program.rs
use solana_sdk::{
    instruction::{AccountMeta, Instruction},
    pubkey::Pubkey,
    signature::Signer,
    transaction::Transaction,
};
fn main() {
    let (client, payer) = solana_client_example::init();
    let program_id: Pubkey = std::env::var("hello_world").unwrap().parse().unwrap();

    let instruction = Instruction {
        program_id,
        accounts: vec![AccountMeta::new(payer.pubkey(), true)],
        data: Vec::new(),
    };
    let transaction = Transaction::new_signed_with_payer(
        &[instruction],
        Some(&payer.pubkey()),
        &[&payer],
        // get_latest_blockhash expired in 2min
        client.get_latest_blockhash().unwrap(),
    );
    match client.send_and_confirm_transaction(&transaction) {
        Ok(signature) => println!("Transaction sent with signature: {}", signature),
        Err(err) => eprintln!("Error sending transaction: {:?}", err),
    }
}

File: typescripts/invoke_program.ts
import {
    PublicKey,
    Transaction,
    sendAndConfirmTransaction,
    TransactionInstruction
} from '@solana/web3.js';
import { init } from './init';
(async () => {
    const { client, payer } = init();
    const programId = new PublicKey(process.env.hello_world!);
    const instruction = new TransactionInstruction({
        keys: [], // our program only logs hello world and doesn't need any accounts.
        programId,
        data: Buffer.alloc(0)
    });
    const signature = await sendAndConfirmTransaction(
        client,
        new Transaction().add(instruction),
        [payer]
    );
    console.log('signature/txhash:', signature);
})();